tickit/adapters/interpreters/command/command_interpreter.py

```diff
@@ -100,3 +100,40 @@ async def handle(
             return resp, command.interrupt
         resp = CommandInterpreter.unknown_command()
         return resp, False
+
+
+class BeheadingInterpreter(Interpreter[AnyStr]):
+    """A decorator for an interpreter which strips a header from a message.
+
+    An interpreter decorator that takes a message, strips off a header of a fixed size
+    and passes the stripped message on to the decorated interpreter
+    """
```

Comment:
Whilst composition could be performed with use of a decorator it is much more likely to be used in a nested manner, e.g.:
```python
BeheadingInterpreter(SomeOtherInterpreter())
```
as opposed to
```python
@BeheadingIntepreter
class MyInterpreter:
    ...
```
As such, I think it would be wise to avoid specifically calling it out as a decorator

Reply:
Is it better to refer to it as a wrapper class?

Reply:
I reckon so, I think that's most likely how it'll be used